packages/upload-core/upload.js

```diff
@@ -117,21 +117,24 @@ class Upload {
   start() {
     const { file } = this;
 
+    const metadata = {
+      'availity-filename': file.name,
+      'availity-content-type': file.type,
+      'availity-attachment-name': 'N/A',
+    };
+    if (this.options.metadata) {
+      Object.assign(metadata, this.options.metadata);
```

Comment:
You could have used the spread operator above:
```js
const metadata = {
  'availity-filename': file.name,
  'availity-content-type': file.type,
  'availity-attachment-name': 'N/A',
  ...this.options.metadata
};
```
If not, there is still no need to do the if check, Object.assign works with falsey values (such as `undefined`, `null`, `false`, and more), it just ignores them.

Reply:
Could not get spread to work with object literals yet https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax and looks like they are not IE compatible. 

Removing the redundant if